### PR TITLE
User story 29/changing pen colour

### LINKED ISSUE #29: As a player, I can change pen colour, so that I can draw more accurately

### PR DESCRIPTION
**Pull request by ChristopherPearse**
Added the ability to change the colour of the pen using various buttons. Created unit tests for included code.

Closes #29

### CODE REVIEW

File: src/public/html/drawingBoard.html
Comment (RudolphTorlage) [inline src/public/html/drawingBoard.html:1]: Added ids to buttons for identification

File: src/public/js/drawingBoard.js
Comment (RudolphTorlage) [inline src/public/js/drawingBoard.js:1]: adhered to all styling guidelines

File: tests/draw.spec.js [hunk 1]
```diff
@@ -35,3 +35,65 @@ test('canvas has white rectangle', async ({ page }) => {
...
+  //Check Pink
+  await page.locator('#pinkButton').click()
+  strokeStyleColor = await page.evaluate(() => {
+    const canvas = document.querySelector('#canvas')
+    const context = canvas.getContext('2d')
+    return context.strokeStyle
+  })
+  expect(strokeStyleColor).toBe('#ffc0cb')
+})
+
```
Comment (RudolphTorlage) [inline tests/draw.spec.js:78]: This test fails, please fix it.
The error seems to stem from trying to click a button after the browser has closed. Possibly due to Firefox running in headless mode.

**Review by RudolphTorlage — CHANGES_REQUESTED** [review]
The multi-color button unit test is failing. I have left a comment explaining what I think the cause might be. Here is a link to said comment
https://github.com/witseie-elen4010/2024-group-lab-004/pull/50/files#r1576336120

Otherwise it looks good.

File: tests/draw.spec.js (hunk 1 above)
Comment (ChristopherPearse) [reply tests/draw.spec.js:78]: I have madechanges and shall redeploy to check tests work now.

Comment (RudolphTorlage) [reply tests/draw.spec.js:78]: thanks

**Review by RudolphTorlage — APPROVED** [review]
Added functionality to change draw colors.
Associated unit tests included.

**Pull request merged**